lightning/src/routing/gossip.rs

```diff
@@ -2121,9 +2214,33 @@ mod tests {
 			Err(e) => assert_eq!(e.err, "Already have chain-validated channel")
 		};
 
+		#[cfg(feature = "std")]
+		{
+			use util::time::tests::MockInstant as Instant;
+			// Insert node 1 into removed_nodes map to track it as removed.
+			gossip_sync.network_graph().removed_nodes.lock().unwrap().insert(NodeId::from_pubkey(
+				&PublicKey::from_secret_key(&secp_ctx, node_1_privkey)), Instant::now());
```

Comment:
Test using the interface by calling `node_failed_permanent` instead. Then below, advance the time and call `remove_stale_channels_and_tracking`.